From f46cbcb53dd94671184350d276ba99348116ddc3 Mon Sep 17 00:00:00 2001
From: Brent Cook <busterb@gmail.com>
Date: Sun, 4 Feb 2024 22:53:59 -0600
Subject: [PATCH 1/2] align read-only sections on masm/windows to 64 bytes

Avoid conflicts where alignment is specified later in the underlying
assembly.
---
 src/lib/libcrypto/perlasm/x86_64-xlate.pl | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/lib/libcrypto/perlasm/x86_64-xlate.pl b/src/lib/libcrypto/perlasm/x86_64-xlate.pl
index 325e7cf58..1e5475312 100755
--- a/src/lib/libcrypto/perlasm/x86_64-xlate.pl
+++ b/src/lib/libcrypto/perlasm/x86_64-xlate.pl
@@ -567,7 +567,15 @@ my %globals;
 					$v.="$line\tSEGMENT";
 					if ($line=~/\.([prx])data/) {
 					    $v.=" READONLY";
-					    $v.=" ALIGN(".($1 eq "p" ? 4 : 8).")" if ($masm>=$masmref);
+					    if ($masm>=$masmref) {
+						    if ($1 eq "r") {
+							    $v.=" ALIGN(64)";
+						    } elsif ($1 eq "p") {
+							    $v.=" ALIGN(4)";
+						    } else {
+							    $v.=" ALIGN(8)";
+						    }
+					    }
 					} elsif ($line=~/\.CRT\$/i) {
 					    $v.=" READONLY ";
 					    $v.=$masm>=$masmref ? "ALIGN(8)" : "DWORD";
-- 
2.43.0

